swap/src/cli/command.rs

```diff
@@ -111,6 +124,64 @@ pub struct MoneroParams {
     pub monero_daemon_host: String,
 }
 
+#[derive(structopt::StructOpt, Debug)]
+pub struct BitcoinParams {
+    #[structopt(long = "electrum-http",
+    help = "Provide the Bitcoin Electrum HTTP URL",
+    default_value = DEFAULT_ELECTRUM_HTTP_URL
+    )]
+    pub electrum_http_url: Url,
+
+    #[structopt(long = "electrum-rpc",
+    help = "Provide the Bitcoin Electrum RPC URL",
+    default_value = DEFAULT_ELECTRUM_RPC_URL
+    )]
+    pub electrum_rpc_url: Url,
+}
+
+#[derive(structopt::StructOpt, Debug)]
+pub struct DataParams {
```

Comment:
You are only using `DataParams` once, can we not just inline that and get rid of the `FromStr` impls etc?

Reply:
yep agreed

Reply:
As discussed: removed `DataParams` in favor of just using `Data` - but the trait impls have to stay for structopt https://docs.rs/structopt/0.3.21/structopt/#default-values - changing the parameter to option instead is not good, because then the help does not properly show the default parameter...